### ISSUE TITLE
Error: read ECONNRESET

### ISSUE THREAD

**Issue by krystianprzybyla** [lukasz-wronski/vscode-ftp-sync#24]
I tried to work with Ovh server (ftp.cluster012.ovh.net). Authorization succeeded but finally (after about 10 seconds) an error appeared:
"Ftp-sync: sync error: Error: read ECONNRESET".

Passive mode was required but "passive": true didn't help.

I spent an hour on modifying settings but no result. Connection to other hosting provider works fine.

I'd appreciate your help!

**Comment by lukasz-wronski** [comment]
Can you create me an FTP account in your server for testing purpouse? If
yes then please send it to the e-mail addres you'd find on plugin homepage.

pt., 15.01.2016, 17:12 krystianprzybyla użytkownik <notifications@github.com>
napisał:

**Comment by lukasz-wronski** [comment]
This is caused by a bug in the ftp library I'm using. I've opened a ticket as above.

**Comment by zer0-day** [comment]
any solution?

**Comment by lukasz-wronski** [comment]
Please follow issue on mscdex/node-ftp referenced above. Without this fix there is nothing i can do.

**Comment by lanouvellecom** [comment]
Hello, 

Thx Microsoft ! A new fantastic web editor and no solution for ftp access or synchronisation ?!
Fortunately, lukasz-wronski your extension is there... but I am also having this same problem. 
It is a shame. I work virtually live on FTP all days and your extension has so much interest to me.

Maybe we could try again to revive the persons concerned and accelerate support of our problem?
I guess not being the only, conditionning my use of this new beautiful visual studio code to this functionality... (not sure that's good in English...).

Thx.

**Comment by wendallsan** [comment]
Would it be feasible to be able to suppress ECONNRESET messages for users that are not having problems with the FTP process, but are still routinely getting this message notice?  In my experience, the plugin is successfully performing all operations, in that it is pushing and fetching files, and the synchronize features are working well, but the ECONNRESET Error notice appears at the top of the IDE every minute or two.  As notices cover the tabs for open files, and has to be constantly closed before a user can interact with these tabs, this becomes a very annoying glitch in an otherwise flawless workflow.

**Comment by inurear** [comment]
I've been getting the same issue with my GoDaddy accounts, however fine with my other accounts. When I turn on the debugger the most I manage is as follows:
`[parser] Response: code=331, buffer='User alex@fm*****.com OK. Password required'
[connection] < '230-Your bandwidth usage is restricted\r\n230 OK. Current restricted directory is /\r\n'
[parser] < '230-Your bandwidth usage is restricted\r\n230 OK. Current restricted directory is /\r\n'
[parser] Response: code=230, buffer='Your bandwidth usage is restricted\r\nOK. Current restricted directory is /'
[connection] > 'FEAT'
[sync-helper] connClosed`

Noting that I have unlimited bandwidth, I am a bit confused to the nature of the error as a response code of 230 means that everything has been authenticated

**Comment by blue875** [comment]
I'm having the same issue, also with GoDaddy FTP.  On my other web host provider (WinHost) I had no such issues. GoDaddy tech support didn't offer up much on my first call. The same credentials work fine with FileZilla and with BeyondCompare...I'm not sure what distinguishes this from the other applications.

**Comment by kalieye** [comment]
if any of you guys is able to provide me a test account with the same problem i will be happy to debug it and try to find a fix

**Comment by blue875** [comment]
@kalieye @lukasz-wronski I have an FTP login I can share, how can I PM you?

**Comment by kalieye** [comment]
@blue875 email me: syn@kalieye.net

**Comment by jmusialek** [comment]
**FIX** 
Hope this helps @blue875 @lukasz-wronski @kalieye @inurear

To fix this issue, edit `node_modules\ftp\lib\connection.js`
Change line 106 from 

`socket.setKeepAlive(true);`
to 
`socket.setKeepAlive(false);`

I had an issue with my godaddy accounts and a similar extension in atom and this was the fix (looks like the exact same core FTP library). I originally found the answer [here](https://github.com/mgrenier/remote-ftp/issues/585) after much searching and pulling of hair!! Thank you @shahafan.

You will need to reload vscode and then it should work like a treat...it did for me.

As this breaks each time there is an update, a permanent fix or config option would be welcomed ;)

**Comment by blue875** [comment]
YES!!!! This fixed it perfectly - made the file change, closed and reopened
VS Code, and it began working immediately.
To be more specific, the file on my PC was in path
C:\Users\[myusername]\.vscode\extensions\lukasz-wronski.ftp-sync-0.3.3\node_modules\ftp\lib\connections.js
And line 106 was the same line in my code file.
Thank you! Not having to bounce between filezilla and VS Code to make
changes between a dev and production site means less opportunity for error.

**Comment by kalieye** [comment]
@jmusialek that's a great find :)  however we cannot fix it because it's part of this: https://github.com/mscdex/node-ftp/issues

maybe you can open an issue there?

Thank you!

**Comment by Fixatron** [comment]
After spending a full day and 6 phone calls to Godaddy, we finally found that the issue was on Godaddy's side. Something was blocking the passive connections.
was told to use the IP address as the host, reset passwords, try different programs. Finally, I told the guy I'm not hanging up until this is solved, and he did it. It was escalated once to a secondary specialist, then he privately contacted another specialist, who solved it in the time it took me to eat my bowl of rice. Specifically came back saying it was an issue with the passive connection (despite me saying so numerous times).
Anyways, great plugin. Too bad godaddy has so many issues (they say its for security but they still want to charge me $200 per domain for a firewall)
